{"bug_class": "PROVER", "kind": "ERROR", "bug_type": "THREAD_SAFETY_VIOLATION", "qualifier": "Unprotected write. Non-private method `org.fim.internal.hash.HashProgress.outputInit` writes to field `this.org.fim.internal.hash.HashProgress.summedFileLength` outside of synchronization.\n Reporting because another access to the same memory occurs on a background thread, although this access may not.", "severity": "HIGH", "visibility": "user", "line": 57, "column": -1, "procedure": "void HashProgress.outputInit()", "procedure_id": "org.fim.internal.hash.HashProgress.outputInit():void.7e204e9b78c1ceadd6ee30598dbf057c", "procedure_start_line": 56, "file": "src/main/java/org/fim/internal/hash/HashProgress.java", "bug_trace": [{"level": 0, "filename": "src/main/java/org/fim/internal/hash/HashProgress.java", "line_number": 57, "column_number": -1, "description": "<Write on unknown thread>"}, {"level": 0, "filename": "src/main/java/org/fim/internal/hash/HashProgress.java", "line_number": 57, "column_number": -1, "description": "access to `this.org.fim.internal.hash.HashProgress.summedFileLength`"}, {"level": 0, "filename": "src/main/java/org/fim/internal/hash/HashProgress.java", "line_number": 71, "column_number": -1, "description": "<Write on background thread>"}, {"level": 0, "filename": "src/main/java/org/fim/internal/hash/HashProgress.java", "line_number": 71, "column_number": -1, "description": "access to `this.org.fim.internal.hash.HashProgress.summedFileLength`"}], "key": "HashProgress.java|outputInit|THREAD_SAFETY_VIOLATION", "node_key": "9c5d6d9028928346cc4fb44cced5dea1", "hash": "c4cf8534f17ac8c79943daf5dea07431", "bug_type_hum": "Thread Safety Violation", "censored_reason": ""}